### PR TITLE
Add tracks fundamentals

### PR DESCRIPTION
**Pull request by kevinyamauchi**
This PR adds a "fundamentals" page describing the API for the new Tracks layer.  We will create an applications page in a follow up PR.

### CODE REVIEW

**Comment by kevinyamauchi** [comment]
Hey @quantumjot, this is ready for a first round of review! If you choose the "view file" option for the markdown file in the "Files Changed" tab (it's in the "..." dropdown by the file header, you will see the markdown rendered. The gifs won't load, but you can view those also using "view file". Looking forward to hearing your thoughts!

File: fundamentals/tracks.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    [3, 4, 636, 1000]
+]
+
+with napari.gui_qt():
+    viewer = napari.view_image(test_im, name='image')
+    viewer.add_tracks(tracks_data, name='tracks')
+
+```
+
+![image]({{ '/assets/tutorials/tracks_simple_demo.gif' | relative_url }})
```
Comment (quantumjot) [inline fundamentals/tracks.md:67]: I love this demo!

File: fundamentals/tracks.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+Returns
+-------
+layer : napari.layers.Tracks
+    The newly-created tracks layer.
+"""
+```
+
+## tracks data
+
+The input data to the tracks layer must be an NxD numpy array
```
Comment (quantumjot) [inline fundamentals/tracks.md:131]: Should we change this now that the `Tracks` layer can accept a list, dataframe or array for `data`?

File: fundamentals/tracks.md [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+Consider a set of 4D tracks data (time + 3 spatial dimensions) defined below. The `Tracks` layer assumes the first column is the `track_id`, the second column is the time axis, and columns 3-5 are Z, Y, and X, respectively. Each row is one vertex in a track. All vertices with the same `track_id` are joined into a single track. In this case, we have defined 2 tracks: track 0, which goes from [10, 10, 10] to [20, 10, 10] and track 1, which goes from [10, 8, 5] to [7, 8, 10] (coordinates written as [x, y z]\).
+
+| track_id | t | z  | y  | x  |
+|----------|---|----|----|----|
+| 0        | 0 | 10 | 10 | 10 |
+| 0        | 1 | 10 | 10 | 20 |
+| 1        | 0 | 5  | 8  | 10 |
+| 1        | 1 | 10 | 8  | 7  |
+
+We can pass the example data above to the tracks layer as follows:
```
Comment (quantumjot) [inline fundamentals/tracks.md:146]:
```suggestion
The data in the array must be sorted by increasing `track_id` then time, as shown above.  We can pass the example data above to the tracks layer as follows:
```

File: fundamentals/tracks.md [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+
+with napari.gui_qt():
+    viewer = napari.view_tracks(tracks_data)
+
+```
+
+
+## tracks graph
+
+We can use the tracks `graph` argument to define the relationships between tracks (e.g., tracks merging or tracks splitting. The graph is defined as a dictionary where the keys are the `track_id` and the values are the `track_id` of the parents of the the key.
```
Comment (quantumjot) [inline fundamentals/tracks.md:165]:
```suggestion
We can use the tracks `graph` argument to define the relationships between tracks (e.g., tracks merging or tracks splitting). The graph is defined as a dictionary where the keys are the `track_id` and the values are the `track_id` of the parents of the the key.
```

File: fundamentals/tracks.md [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+    1: [0],
+    2: [0],
+    3: [1, 2]  
+}
+```
+For a full example of 3d+t tracks data with a parent graph, please see our [`tracks_3d_with_graph.py` example](https://github.com/napari/napari/blob/master/examples/tracks_3d_with_graph.py).
+
+## using the tracks properties dictionary
+
+The `Tracks` layer can contain properties that annotate each point.
```
Comment (quantumjot) [inline fundamentals/tracks.md:188]:
```suggestion
The `Tracks` layer can contain properties that annotate the vertices of each track.
```

**Review by quantumjot — COMMENTED** [review]
Hi @kevinyamauchi - I think this looks really great! I've made a few small suggestions and comments.

File: fundamentals/tracks.md (hunk 2 above)
Comment (kevinyamauchi) [reply fundamentals/tracks.md]: Great point. Fixed!

**Comment by kevinyamauchi** [comment]
Thanks for the speedy review, @quantumjot ! I have addressed your comments.

This PR is ready for review, @napari/core-devs !

**Comment by kevinyamauchi** [comment]
Hey @quantumjot , the btrack example data you linked looks like it would work nicely for the demo! How large is the corresponding image file? I think it would be cool to include images of the cells being tracked, but I understand if it's not possible to share in a convenient way.

**Review by sofroniewn — APPROVED** [review]
This is great @kevinyamauchi, love the demos!! Thanks for the review too @quantumjot. I'm merging now, looking forward to the more advanced tutorial too!!

**Comment by kevinyamauchi** [comment]
Great! Thanks @quantumjot and @sofroniewn !

**Pull request merged**